src/handeye_calibration.cpp

```diff
@@ -82,41 +85,41 @@ int readTransformPairsFromFile(std::string filename, EigenAffineVector& t1, Eige
 	auto t2_it = std::back_inserter(t2);
 
 	if(fs.isOpened())
-	{
-
-		for (int i = 0; i < frameCount; ++i, ++t1_it, ++t2_it){
-			// read in frame one
-			{
-				cv::Mat_<double> t1cv = cv::Mat_<double>::ones(4,4);
-				std::stringstream ss1;
-				ss1 << "T1_" << i;
-				fs[ss1.str()] >> t1cv;
-				Eigen::Affine3d t1e;
-				cv::cv2eigen(t1cv,t1e.matrix());
-				*t1_it = t1e;
-			}
-
-			// read in frame two
-			{
-				cv::Mat_<double> t2cv = cv::Mat_<double>::ones(4,4);
-				std::stringstream ss2;
-				ss2 << "T2_" << i;
-				fs[ss2.str()] >> t2cv;
-				Eigen::Affine3d t2e;
-				cv::cv2eigen(t2cv,t2e.matrix());
-				*t2_it = t2e;
-			}
-
-		}
-		fs.release();
-	} else {
+    {
+
+      for (int i = 0; i < frameCount; ++i, ++t1_it, ++t2_it){
+        // read in frame one
+        {
+          cv::Mat_<double> t1cv = cv::Mat_<double>::ones(4,4);
+          std::stringstream ss1;
+          ss1 << "T1_" << i;
+          fs[ss1.str()] >> t1cv;
+          Eigen::Affine3d t1e;
+          cv::cv2eigen(t1cv,t1e.matrix());
+          *t1_it = t1e;
+        }
+
+        // read in frame two
+        {
+          cv::Mat_<double> t2cv = cv::Mat_<double>::ones(4,4);
+          std::stringstream ss2;
+          ss2 << "T2_" << i;
+          fs[ss2.str()] >> t2cv;
+          Eigen::Affine3d t2e;
+          cv::cv2eigen(t2cv,t2e.matrix());
+          *t2_it = t2e;
+        }
+
+      }
+      fs.release();
+    } else {
 		std::cerr << "failed to open input file " << filename << "\n";
 		return 1;
 	}
 	return 0;
 }
 
-Eigen::Affine3d estimateHandEye(const EigenAffineVector& baseToTip, const EigenAffineVector& camToTag)
+Eigen::Affine3d estimateHandEye(const EigenAffineVector& baseToTip, const EigenAffineVector& camToTag, ceres::Solver::Summary& summary)
```

Comment:
I think the summary info is great to have, but can the summary be an optional parameter?

I'd like to keep this function as easy to use as possible in the most basic case.

Reply:
I'll try to look into how to do that in C++ over the wee-end.

Reply:
I think just adding an overload (another function definition) without it that creates an unused one internally and then calls this version would be fine.